contentcuration/contentcuration/static/js/edit_channel/uploader/views.js

```diff
@@ -10,164 +10,164 @@ var get_cookie = require("utils/get_cookie");
-			callback: this.create_exercise,
-			modal: true,
-			parent_view: this,
-			model: this.model
-		});
-	},
-	create_exercise:function(exercise_list){
-		console.log("exercises", exercise_list);
-	},
-	close: function() {
+    /*Render any items that have previously been added*/
+    load_content:function(){
+        var self = this;
+        if(this.collection){
+            this.collection.forEach(function(entry){
+                var node_view = new NodeListItem({
+                    containing_list_view: self,
+                    el: self.$el.find("#content_item_" + entry.cid),
+                    model: entry,
+                    root: this.model
+                });
+                self.views.push(node_view);
+            });
+        }
+    },
+    add_topic:function(){
+        var topic = new Models.NodeModel({
+            "kind":"topic",
+            "title": (this.counter > 0)? "Topic " + this.counter : "Topic",
+            "parent" : this.model.id,
+            "sort_order" : this.main_collection.length + this.collection.length
+        });
+        this.collection.add(topic);
+        this.counter++;
+        var item_view = new NodeListItem({
+            containing_list_view: this,
+            model: topic,
+            root: this.model
+        });
+        $("#upload_content_add_list").append(item_view.el);
+        this.views.push(item_view);
+    },
+    edit_metadata: function(){
+        if(this.modal){
+            this.$el.find(".modal").modal("hide");
+        }else{
+            this.close();
+        }
+        $("#main-content-area").append("<div id='dialog'></div>");
+        var metadata_view = new EditMetadataView({
+            el : $("#dialog"),
+            collection: this.collection,
+            parent_view: this.parent_view,
+            model: this.model,
+            allow_add: true,
+            main_collection : this.main_collection,
+            modal: true
+        });
+    },
+    add_file:function(){
+        var view = new FileUploadView({
+            callback: this.upload_file,
+            modal: true,
+            parent_view: this
+        });
+    },
+    upload_file:function(file_list){
+        console.log("uploading files", file_list);
+        var self = this.parent_view;
+
+        file_list.forEach(function(entry){
+            console.log("this is now", self);
+            var type = "";
+            var file = entry.data;
+            var filename = entry.filename;
+            if(file.type.indexOf("image") >=0){
+                type = "image";
+            }
+            else if(file.type.indexOf("application/pdf") >=0){
+                type="pdf";
+            }
+            else if(file.type.indexOf("text") >=0){
+                type="text";
+            }
+            else if(file.type.indexOf("audio") >=0){
+                type="audio";
+            }
+            else if(file.type.indexOf("video") >=0){
+                type="video";
+            }
```

Comment:
the if - else if - else if ... pattern here seems better off to use a switch statement